jni/include/nmslib_stream_support.h

```diff
@@ -44,6 +44,27 @@ class NmslibOpenSearchIOReader final : public similarity::NmslibIOReader {
 };  // class NmslibOpenSearchIOReader
 
 
+class NmslibOpenSearchIOWriter final : public similarity::NmslibIOWriter {
+ public:
+  explicit NmslibOpenSearchIOWriter(NativeEngineIndexOutputMediator *_mediator)
+      : similarity::NmslibIOWriter(),
+        mediator(_mediator) {
```

Comment:
What if mediator is null here , how we are handling this?

Reply:
Mediator will be passed from wrapper, why it can be null?

Reply:
I see and are we making sure that does wraper always gurantee that it won't pass null